envelope/envelope.go

```diff
@@ -48,3 +52,31 @@ func Write(p protocol.Protocol, w io.Writer, seqID int32, e Enveloper) error {
 		Value: body,
 	}, w)
 }
+
+// ReadReply reads an envelope from the given reader, and decodes any
+// Exception paylods.
```

Comment:
Nit: Decoding exception payloads is an implementation detail. Just mentioning that we're reading enveloped responses should be enough.